### PR TITLE
Fix subscribeQueue issue with empty policy

### PR DESCRIPTION
**Pull request by chidoli**
Fix the issue that subscribeQueue with extendPolicy cannot make subscription when there is no existing policy.

### CODE REVIEW

File: aws-java-sdk-sns/src/main/java/com/amazonaws/services/sns/util/Topics.java [hunk 1]
```diff
@@ -186,8 +186,9 @@ public static String subscribeQueue(AmazonSNS sns, AmazonSQS sqs, String snsTopi
         Map<String, String> sqsAttrs = sqs.getQueueAttributes(sqsQueueUrl, sqsAttrNames).getAttributes();
         String sqsQueueArn = sqsAttrs.get(QueueAttributeName.QueueArn.toString());
 
-        Policy policy = extendPolicy ? Policy.fromJson(
-                sqsAttrs.get(QueueAttributeName.Policy.toString())) : new Policy();
+        String policyJson = sqsAttrs.get(QueueAttributeName.Policy.toString());
+        Policy policy = extendPolicy || policyJson == null || policyJson.length() == 0
```
Comment (breedloj) [inline aws-java-sdk-sns/src/main/java/com/amazonaws/services/sns/util/Topics.java]: Maybe I'm missing something here but don't you only want to extend the policy if the flag is set, policyJson is NOT null and NOT empty?

Comment (chidoli) [inline aws-java-sdk-sns/src/main/java/com/amazonaws/services/sns/util/Topics.java]: I have several environments and I guess I got confused. I will correct it. Thanks a lot.

**Comment by breedloj** [comment]
Change looks good.  Can you confirm you are submitting this under the Apache 2.0 license?

**Comment by chidoli** [comment]
Yes

**Comment by breedloj** [comment]
This has been staged for release and should be available shortly.

**Pull request closed without merging**